src/persist-txn/src/txn_cache.rs

```diff
@@ -91,6 +91,12 @@ pub struct TxnsCacheState<T: Timestamp + Lattice + Codec64> {
     batch_idx: HashMap<Vec<u8>, usize>,
     /// The times at which each data shard has been written.
     pub(crate) datas: BTreeMap<ShardId, DataTimes<T>>,
+    /// The registers needing application as of the current progress.
+    ///
+    /// Invariant: Values are sorted by timestamp.
```

Comment:
These should all be either `>=` or `>` (I think former?) the `self.since_ts`, too, yeah?

Reply:
Yes good point, I think that's true of `unnapplied_batches` too right?

Reply:
Oh I see, it's already in the comment above `since_ts`.